src/routes/api-documentation.ts

```diff
@@ -1,140 +1,141 @@
+      quad(collectionGet, hydra.title, literal('Get the collection', 'en')),
+      quad(collectionGet, hydra.method, literal('GET')),
+      quad(collectionGet, hydra.expects, owl.Nothing),
+      quad(collectionGet, hydra.returns, hydra.Collection),
+      quad(hydra.Collection, hydra.supportedOperation, collectionAdd),
+      quad(collectionAdd, rdf.type, hydra.Operation),
+      quad(collectionAdd, hydra.title, literal('Add an article', 'en')),
+      quad(collectionAdd, hydra.method, literal('POST')),
+      quad(collectionAdd, hydra.expects, schema.Article),
+      quad(collectionAdd, hydra.returns, owl.Nothing),
+      quad(collectionAdd, hydra.possibleStatus, collectionAdd200),
+      quad(collectionAdd200, hydra.statusCode, toRdf(constants.HTTP_STATUS_NO_CONTENT)),
+      quad(collectionAdd200, hydra.description, literal('If the article was added successfully', 'en')),
+      quad(hydra.Collection, hydra.supportedProperty, collectionName),
+      quad(collectionName, rdf.type, hydra.SupportedProperty),
+      quad(collectionName, hydra.title, literal('Title', 'en')),
+      quad(collectionName, hydra.property, hydra.title),
+      quad(hydra.title, rdf.type, rdf.Property),
+      quad(collectionName, hydra.required, toRdf(true)),
+      quad(collectionName, hydra.readable, toRdf(true)),
+      quad(collectionName, hydra.writeable, toRdf(false)),
+      quad(hydra.Collection, hydra.supportedProperty, collectionManages),
+      quad(collectionManages, rdf.type, hydra.SupportedProperty),
+      quad(collectionManages, hydra.title, literal('What the collection manages', 'en')),
+      quad(collectionManages, hydra.property, hydra.manages),
+      quad(collectionManages, hydra.required, toRdf(true)),
+      quad(collectionManages, hydra.readable, toRdf(true)),
+      quad(collectionManages, hydra.writeable, toRdf(false)),
+      quad(hydra.Collection, hydra.supportedProperty, collectionTotalItems),
+      quad(collectionTotalItems, rdf.type, hydra.SupportedProperty),
+      quad(collectionTotalItems, hydra.title, literal('Total items', 'en')),
+      quad(collectionTotalItems, hydra.property, hydra.totalItems),
+      quad(hydra.totalItems, rdf.type, rdf.Property),
+      quad(collectionTotalItems, hydra.required, toRdf(true)),
+      quad(collectionTotalItems, hydra.readable, toRdf(true)),
+      quad(collectionTotalItems, hydra.writeable, toRdf(false)),
+      quad(hydra.Collection, hydra.supportedProperty, collectionMember),
+      quad(collectionMember, rdf.type, hydra.SupportedProperty),
+      quad(collectionMember, hydra.title, literal('Members of this collection', 'en')),
+      quad(collectionMember, hydra.property, hydra.member),
+      quad(hydra.member, rdf.type, rdf.Property),
+      quad(hydra.member, rdf.type, hydra.Link),
+      quad(collectionMember, hydra.required, toRdf(true)),
+      quad(collectionMember, hydra.readable, toRdf(false)),
+      quad(collectionMember, hydra.writeable, toRdf(false)),
+    );
```

Comment:
See also https://github.com/libero/article-store/pull/64#discussion_r352024083.

Reply:
I do simply write all my API Documentation in turtle and parse at app start.

I see you already have `clownface` here. This could become more approachable

```js
import cf = require('clownface')
import dataset = require('rdf-dataset-indexed')

const apiDocumentation = cf({
  dataset: dataset(),
  term: namedNode(url.resolve(request.origin, router.url(Routes.ApiDocumentation)))
})

// individual triples are boring
apiDocumentation.addOut(rdf.type, hydra.ApiDocumentation)
apiDocumentation.addOut(hydra.entrypoint, entryPoint)

// but you can create deep graphs with callbacks
apiDocumentation.addOut(hydra.supportedClass, schema.EntryPoint, entrypoint => {
  entrypoint.addOut(rdf.type, hydra.Class)
  entrypoint.addOut(hydra.title, literal('API entry point', 'en'))

  entrypoint.addOut(rdf.type, hydra.supportedOperation, entryPointGet => {
    entryPointGet.addOut(rdf.type, hydra.Operation)
  })
})
```

Reply:
Interesting, will have a go. 👍

Reply:
The nesting via callbacks is interesting here in terms of readability.